rotkehlchen/tests/unit/test_general_cache.py

```diff
@@ -0,0 +1,104 @@
+from unittest.mock import patch
+
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.globaldb import GlobalDBHandler
+from rotkehlchen.types import GeneralCacheType
+
+
+def test_curve_pools_cache(rotkehlchen_instance):
+    """Test curve pools fetching mechanism"""
+
+    # Set initial cache data to check that it is gone after the cache update
+    GlobalDBHandler().delete_general_cache(cache_type=GeneralCacheType.CURVE_POOL_ADDR)
+    GlobalDBHandler().delete_general_cache(cache_type=GeneralCacheType.CURVE_POOL_TOKENS)
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        GlobalDBHandler().set_general_cache_value(
+            write_cursor=write_cursor,
+            cache_type=GeneralCacheType.CURVE_POOL_ADDR,
+            name='abc',
+            value='def',
+        )
+        GlobalDBHandler().set_general_cache_value(
+            write_cursor=write_cursor,
+            cache_type=GeneralCacheType.CURVE_POOL_TOKENS,
+            name='klm',
+            value='xyz',
+        )
+
+    # delete one of the tokens to check that it is created during the update
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        try:
+            GlobalDBHandler().delete_ethereum_token(
+                write_cursor=write_cursor,
+                address='0xD71eCFF9342A5Ced620049e616c5035F1dB98620',
+            )
+        except InputError:
+            # token might not exist but we don't care
+            pass
+
+    # check that it was deleted successfully
+    token = GlobalDBHandler().get_ethereum_token('0xD71eCFF9342A5Ced620049e616c5035F1dB98620')
+    assert token is None
+
+    def mock_call_contract(contract, ethereum, method_name, **kwargs):
+        if method_name == 'pool_count':
+            return 2  # if we don't limit pools count, the test will run for too long
+        return ethereum.call_contract(
+            contract_address=contract.address,
+            abi=contract.abi,
+            method_name=method_name,
+            **kwargs,
+        )
+
+    call_contract_patch = patch(
+        target='rotkehlchen.chain.ethereum.contracts.EthereumContract.call',
+        new=mock_call_contract,
+    )
+
+    with call_contract_patch:
+        rotkehlchen_instance.update_curve_pools_cache()
+
+    expected_lp_tokens_to_pools = {
```

Comment:
Maybe keep this mapping as constants outside of the function, what do you think?

Reply:
Hm... not sure actually since it is not used anywhere else

Reply:
Keep big constants outside. This is what we do in other tests and helps keep the logic of the tests clean and easier to understand